src/types.ts

```diff
@@ -6,6 +8,10 @@ export type KeyMultiValue = {
   [name: string]: string[]
 };
 
+export interface AwsContext extends Context {
+  aws: { context: any, event: AwsRequest }
+}
+
```

Comment:
I don't think we have to literally extend it `Context`. I would rather use interface declaration merging.

The syntax for this would be something like this (untested, based on memory):

```typescript
declare module "@curveball/core" {
  interface Context {
    aws?: { 
      context: any,
      event: any
    }
}
```

The issue with simply extending the interface, is that everything else needs to be extended (Application, middlewares, etc), or add typeguards in a bunch of different places. Neither are ideal I think.

by using declaration merging, all you have to do is install the package, and `ctx.aws` will be immediately available everywhere.

Reply:
I just read up on declaration merging. Interesting stuff to have in your toolbox :)

Reply:
Yea =) makes total sense too for a language where monkey patching is common